### PR TITLE
Update provider versions and tests

### PR DESCRIPTION
**Pull request by nstogner**
Still needs to be tested.

Should non-beta google provider version be bumped as well?

### CODE REVIEW

**Comment by morgante** [comment]
@nstogner Yes let's bump the versions on both to the latest release.

**Comment by nstogner** [comment]
- Refactored tests to use kitchen terraform rather than bats.
- Added terraform module for the base resources being tested.

File: README.md [hunk 1]
```diff
@@ -211,15 +214,15 @@ The project has the following folders and files:
...
+
 ### Terraform plugins
-- [Terraform](https://www.terraform.io/downloads.html) 0.10.x
-- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) 1.10.0
 
-### jq
-- [jq](https://stedolan.github.io/jq/) 1.5
-- [terraform-docs](https://github.com/segmentio/terraform-docs/releases) 0.3.0
+- [Terraform](https://www.terraform.io/downloads.html) 0.11.x
+- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) 1.10.0
```
Comment (aaron-lane) [inline README.md]:
```suggestion
- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) 1.20.X
```

File: README.md [hunk 2]
```diff
@@ -211,15 +214,15 @@ The project has the following folders and files:
...
+- [terraform-provider-google-beta](https://github.com/terraform-providers/terraform-provider-google-beta) 1.19.X
```
Comment (aaron-lane) [inline README.md]:
```suggestion
- [terraform-provider-google-beta](https://github.com/terraform-providers/terraform-provider-google-beta) 1.20.X
```

File: test/integration/full/controls/gcloud.rb [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+          res = JSON.parse(subject.stdout, symbolize_names: true)
+          # TODO: Assert before drilling into json.
+          bindings = res[:bindings]
+          bindings.find { |b| b[:'role'] == expectedRole }[:members]
+        else
+          {}
+        end
+      end
+  
+      it { expect(members).to include(*expectedMembers) }
```
Comment (aaron-lane) [inline test/integration/full/controls/gcloud.rb]:
```suggestion
      it { expect(stdout).to include bindings: including(role: expected_role, members: expected_members) }
```

**Comment by nstogner** [comment]
@aaron-lane The original purpose of this PR was to add in the `google-beta` provider. In order to run tests, there were a good number of resources needed. I switched the test framework from `bats` to `kitchen` to incorporate the creation of the base resources into the codebase (to ease testing). I am not 100% sure on why the resource pairs were there in the beginning, I was hoping to make this PR as close to 1:1 with the original tests as possible so I kept the pairs.

**Pull request merged**